File: analytics/metrics-platform/src/main/java/org/wikimedia/metrics_platform/config/sampling/SampleConfig.java
package org.wikimedia.metrics_platform.config.sampling;

import com.google.gson.annotations.SerializedName;

public class SampleConfig {

    public enum Identifier {
        @SerializedName("session") SESSION,
        @SerializedName("device") DEVICE,
        @SerializedName("pageview") PAGEVIEW
    }

    /** Sampling rate. **/
    double rate;

    /** ID type to use for sampling. */
    Identifier identifier;
}

File: analytics/metrics-platform/src/main/java/org/wikimedia/metrics_platform/config/sampling/SampleConfig.kt
package org.wikimedia.metrics_platform.config.sampling

import kotlinx.serialization.Serializable

@Serializable
class SampleConfig(
    val rate: Double = 1.0,
    val unit: String = UNIT_SESSION
) {
    companion object {
        const val UNIT_PAGEVIEW = "pageview"
        const val UNIT_SESSION = "session"
        const val UNIT_DEVICE = "device"
    }
}
